src/adapters/interlay.ts

```diff
@@ -239,16 +239,28 @@ class InterlayBalanceAdapter extends BalanceAdapter {
 
     return this.storages.assets(address, tokenId).observable.pipe(
       map((balance) => {
-        const amount = FN.fromInner(
+        const free = FN.fromInner(
           balance.free?.toString() || "0",
           this.getToken(token).decimals
         );
 
+        const frozen = FN.fromInner(
+          balance.frozen?.toString() || "0",
+          this.getToken(token).decimals
+        );
+
+        const reserved = FN.fromInner(
+          balance.reserved?.toString() || "0",
+          this.getToken(token).decimals
+        );
+
+        const available = free.sub(frozen);
+
         return {
-          free: amount,
-          locked: new FN(0),
-          reserved: new FN(0),
-          available: amount,
+          free,
+          locked: frozen,
+          reserved,
+          available,
```

Comment:
I'd be tempted to name this `transferable` to match the lib

Reply:
> I'd be tempted to name this `transferable` to match the lib

These names are the same across other adapters (some of which use other pallets to query balances) to match the shared `BalanceData` interface defined [here](https://github.com/interlay/bridge/blob/dc945a97dce4e5ee6019d04a9a2173c495338303/src/types.ts#L121-L126)

I'll keep it in mind, maybe a rename makes sense across the board.